### ISSUE TITLE
constant checks

### ISSUE THREAD

**Issue by staabm** [maglnet/ComposerRequireChecker#356]
with code like

```php
		if (defined('CURLOPT_SSL_VERIFYHOST') && $curlOpt === CURLOPT_SSL_VERIFYHOST) {
			return new UnionType([new ConstantIntegerType(0), new ConstantIntegerType(2)]);
		}
```

the tool properly detects and reports a error:

```sh
php build/composer-require-checker.phar check --config-file /home/runner/work/phpstan-src/phpstan-src/build/composer-require-checker.json
ComposerRequireChecker 4.0.0@baa11a4e9e5072117e3d180ef16c07036cafa4a2
The following 1 unknown symbols were found:
+------------------------+--------------------+
| Unknown Symbol         | Guessed Dependency |
+------------------------+--------------------+
| CURLOPT_SSL_VERIFYHOST | ext-curl           |
+------------------------+--------------------+
```

doing the same within a loop in a "more dynamic" way does not report any errors:

```php
		$boolConstants = [
			'CURLOPT_AUTOREFERER',
			'CURLOPT_COOKIESESSION',
		];
		foreach ($boolConstants as $constName) {
			if (defined($constName) && constant($constName) === $curlOpt) {
				return new BooleanType();
			}
		}
```

I think - for consistency it would be helpful if the require checker could detect such constant cases.

found this inconsistency while working on https://github.com/phpstan/phpstan-src/pull/1719/

**Comment by staabm** [comment]
looking at 

```
		if (defined('CURLOPT_SSL_VERIFYHOST') && $curlOpt === CURLOPT_SSL_VERIFYHOST) {
			return new UnionType([new ConstantIntegerType(0), new ConstantIntegerType(2)]);
		}
```

again, I even think, that the constant should not be reported at all, since the IF construction makes it a optional dependency

**Comment by Ocramius** [comment]
I would rather say that the dependency should be declared, so that it cannot be a constant polyfilled incorrectly by a third party component.

That said, this component uses dirt dumb AST traversal: it doesn't evaluate expressions at all.
